src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/SocketsHttpHandler.cs

```diff
@@ -0,0 +1,238 @@
+        }
+
+        public int MaxResponseDrainSize
+        {
+            get => throw new PlatformNotSupportedException("Property MaxResponseDrainSize is not supported.");
+            set => throw new PlatformNotSupportedException("Property MaxResponseDrainSize is not supported.");
+        }
+
+        public TimeSpan ResponseDrainTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property ResponseDrainTimeout is not supported.");
+            set => throw new PlatformNotSupportedException("Property ResponseDrainTimeout is not supported.");
+        }
+
+        public int MaxResponseHeadersLength
+        {
+            get => throw new PlatformNotSupportedException("Property MaxResponseHeadersLength is not supported.");
+            set => throw new PlatformNotSupportedException("Property MaxResponseHeadersLength is not supported.");
+        }
+
+        [AllowNull]
+        public SslClientAuthenticationOptions SslOptions
+        {
+            get => throw new PlatformNotSupportedException("Property SslOptions is not supported.");
+            set => throw new PlatformNotSupportedException("Property SslOptions is not supported.");
+        }
+
+        public TimeSpan PooledConnectionLifetime
+        {
+            get => throw new PlatformNotSupportedException("Property PooledConnectionLifetime is not supported.");
+            set => throw new PlatformNotSupportedException("Property PooledConnectionLifetime is not supported.");
+        }
+
+        public TimeSpan PooledConnectionIdleTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property PooledConnectionLifetime is not supported.");
+            set => throw new PlatformNotSupportedException("Property PooledConnectionLifetime is not supported.");
+        }
+
+        public TimeSpan ConnectTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property ConnectTimeout is not supported.");
+            set => throw new PlatformNotSupportedException("Property ConnectTimeout is not supported.");
+        }
+
+        public TimeSpan Expect100ContinueTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property Expect100ContinueTimeout is not supported.");
+            set => throw new PlatformNotSupportedException("Property Expect100ContinueTimeout is not supported.");
+        }
+
+        public IDictionary<string, object?> Properties => throw new PlatformNotSupportedException("Property Properties is not supported.");
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing && !_disposed)
+            {
+                _disposed = true;
+                _handler?.Dispose();
+                _handler = null;
+            }
+
+            base.Dispose(disposing);
+        }
+
+        private HttpMessageHandler SetupHandlerChain()
```

Comment:
We shouldn't be replacing the SocketsHttpHandler implementation.  We should be replacing the HttpClientHandler implementation.  If we want to replace SocketsHttpHandler as part of this change, it can be to replace the whole thing with throwing PNSEs when built for the browser. SocketsHttpHandler is logically a peer to WinHttpHandler, CurlHandler, BrowserHttpHandler, AndroidHandler, etc... they each represent an implementation tied to a specific set of underlying APIs... HttpClientHandler is the generic handler that's created by default by HttpClient, and it's the one we plumb to instantiate the correct underlying implementation.

Reply:
The reason was so that we could `throw new PlatformNotSupportedException(` in the properties.  Looking at `HttpConnectionSettings` it does not really lend itself to throwing the PNSE as it returns field in `SocketsHttpHandler`.

Reply:
What we can do is use a preprocessor and move the PlatformNotSupportedException to those properties.

Reply:
This has been addressed.